pkg/auth/providers/activedirectory/activedirectory_client.go

```diff
@@ -163,18 +210,18 @@ func (p *adProvider) getPrincipalsFromSearchResult(result *ldapv2.SearchResult,
 				}
 				groupList = append(groupList, grp)
 			}
-			return userPrincipal, groupList, nil
+			return groupList, nil
 		}
-		return userPrincipal, groupPrincipals, err
+		return groupPrincipals, err
 	}
 
-	result, err = lConn.SearchWithPaging(search, 1000)
+	result, err := lConn.SearchWithPaging(search, 1000)
 	if err != nil {
 		// check errors
 		if ldapErr, ok := err.(*ldapv2.Error); ok && ldapErr.ResultCode == 32 {
-			return userPrincipal, groupPrincipals, err
+			return groupPrincipals, err
```

Comment:
We seem to be handling ldapErr.ResultCode == 32 by returning a "Not Found" httperror elsewhere. Does this apply here? Or should we just return the err as-is?